Reed/DefinitionView/viewmodels/DefinitionViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  DefinitionModalViewModel.swift
+//  Reed
+//
+//  Created by Hugo Zhan on 11/3/20.
+//  Copyright © 2020 Roger Luo. All rights reserved.
+//
+
+import SwiftUI
+
+class DefinitionViewModel: ObservableObject {
+    @Published var title = ""
+    @Published var currReading = ""
+    @Published var terms = [Terms]()
+    @Published var definitons = [Defs]()
+    
+    let dictionaryFetcher = DictionaryFetcher()
+    
+    func generateContent(from tappedWord: String) {
+        title = tappedWord
+        var termsArr = [Terms]()
+        var defsArr = [Defs]()
+        let entries = dictionaryFetcher.fetchEntries(of: tappedWord)
+        if entries.endIndex > 0
+            && entries[0].readings[0].terms.endIndex > 0
+            && entries[0].readings[0].terms[0] == tappedWord {
+            currReading = entries[0].readings[0].reading
+        } else {
+            currReading = ""
+        }
+        for entry in entries {
+            for reading in entry.readings {
+                for term in reading.terms {
+                    termsArr.append(Terms(reading: reading.reading, term: term))
+                }
+            }
+            if currReading != "" {
+                termsArr.removeFirst()
+            }
+            for sense in entry.definitions {
+                var definition = ""
+                for gloss in sense.glosses {
+                    definition += gloss + ", "
+                }
+                definition.removeLast(2)
+                var specicificLexemes = ""
+                if !sense.specificLexemes.isEmpty {
+                    for specific in sense.specificLexemes {
+                        specicificLexemes += specific + ", "
+                    }
+                    specicificLexemes.removeLast(2)
+                }
+                defsArr.append(Defs(specicificLexemes: specicificLexemes, definition: definition))
+            }
+        }
+        terms = termsArr
+        definitons = defsArr
```

Comment:
Change `termsArr` to `terms` and use `self.terms = terms` here. Same for `defsArr`.